src/Features/WetnessEffects.h

```diff
@@ -66,4 +93,32 @@ struct WetnessEffects : Feature
 	virtual void RestoreDefaultSettings();
 	float CalculateWeatherTransitionPercentage(float skyCurrentWeatherPct, float beginFade, bool fadeIn);
 	void CalculateWetness(RE::TESWeather* weather, RE::Sky* sky, float seconds, float& wetness, float& puddleWetness);
+
+	virtual void PostPostLoad() override { Hooks::Install(); }
+
+	struct Hooks
+	{
+		struct BSParticleShader_SetupGeometry
+		{
+			static void thunk(RE::BSShader* This, RE::BSRenderPass* Pass, uint32_t RenderFlags)
+			{
+				func(This, Pass, RenderFlags);
+
+				auto particleShaderProperty = (RE::BSParticleShaderProperty*)Pass->shaderProperty;
+				auto cube = (RE::BSParticleShaderCubeEmitter*)particleShaderProperty->particleEmitter;
+				GetSingleton()->precipProj = cube->occlusionProjection;
```

Comment:
Ok, I think if we do a nullptr check this should still work. I'm not sure if we have to exit early though when that happens.
```cpp
				if (cube)
					GetSingleton()->precipProj = cube->occlusionProjection;
```